### PR TITLE
Add utility library to manipulate net-attach-def and annotation

### PR DESCRIPTION
**Pull request by s1061123**
This change introduce net-attach-def utility library and updated
networkSelectionElement to support upcoming specification.

### CODE REVIEW

**Comment by phoracek** [comment]
@s1061123 whoah, looks great. I like to see all those helpers and well formulated fields. Would you please share link to the "upcoming specification"?

File: pkg/utils/net-attach-def.go [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+// limitations under the License.
+
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+	//"k8s.io/client-go/rest"
```
Comment (phoracek) [inline pkg/utils/net-attach-def.go]: Please drop unused imports. There are more of them in this and other added modules.

Comment (s1061123) [reply pkg/utils/net-attach-def.go]: fixed!

**Comment by s1061123** [comment]
@phoracek Newer spec is accepted in NPWG and it will be uploaded soon at https://github.com/k8snetworkplumbingwg/multi-net-spec

For now, you can see 'V1.1 spec' in following minutes/agenda doc.
https://docs.google.com/document/d/1oE93V3SgOGWJ4O1zeD1UmpeToa0ZiiO6LqRAmZBPFWM/edit

**Comment by s1061123** [comment]
@dcbw @dougbtv PTAL?

**Comment by dcbw** [comment]
FWIW looks good to me too

**Pull request merged**